specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-05-01/managedClusters.json

```diff
@@ -3317,6 +3320,10 @@
         "privateDNSZone": {
           "type": "string",
           "description": "Private dns zone mode for private cluster. "
+        },
+        "enablePrivateClusterPublicFQDN": {
+          "type": "boolean",
```

Comment:
ARM generally discourages boolean properties because true/false values are not helpful to understand their meaning. Recommendation is to choose a property that is a natural enum, like additionalFqdnType: { Public, None }. This is usually easier for users of the API to understand. If true/false values are still preferred, consider using enum { True, False }. Both of these approaches allow adding enum values without a new API version if they they are declared with x-ms-enum { modealAsString: true }.

Reply:
The property is already defined inside our RP code as bool. If I change this to enum, would it cause issue during unmarshal because it's type is changed from string to bool?
As far as I know, if request body is sent with `enablePrivateClusterPublicFQDN: "True"`, it will fail at our RP side.

Reply:
This is a new property you are adding, and just now publishing, so your RP can implement it however. ARM review isn't just about making these specs conform to what your code already does. It's to give your whole team/RP feedback about its API.

Reply:
I won't block approval on this change: ultimately it's your RP's decision. And it's not just me that discourages booleans, it's a general ARM recommendation: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean%20enum#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean%20enum#common-issues-found-in-review).

I didn't say booleans are confusing, but I'm saying well-chosen enums are an overall better API design and it's part of creating more consistency across all of Azure APIs to help our customers. Enum values convey information about what they are by the choice of the value itself (e.g. Public). True and False only tell whether a switch is on or off, nothing about what the switch controls. Also, enums are extensible and booleans are not.

I like your idea of examining the API overall for these types of improvements in a future API version. Please take this input and other guidelines in the wiki into consideration when you do so.

Reply:
Thanks @mentat9